File: src/game/powerups/LivesPowerUp.java
package game.powerups;

import game.data.DataInputOutput;
import game.entity.models.BomberEntity;
import game.models.Coordinates;
import game.utils.Paths;

import javax.xml.crypto.Data;
import java.awt.image.BufferedImage;

public class LivesPowerUp extends PowerUp{
    /**
     * Constructs a PowerUp entity with the specified coordinates.
     *
     * @param coordinates the coordinates of the PowerUp entity
     */
    public LivesPowerUp(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.getPowerUpsFolder() + "/lives_up.png");
    }

    @Override
    public int getDuration() {
        return 0;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        DataInputOutput.increaseLives();
    }

    @Override
    protected void cancel(BomberEntity entity) {

    }
}

File: src/game/powerups/LivesPowerUp.kt
package game.powerups

import game.data.DataInputOutput
import game.entity.EntityTypes
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.utils.Paths.powerUpsFolder
import java.awt.image.BufferedImage
class LivesPowerUp : PowerUp {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override fun getImage(): BufferedImage =
            loadAndSetImage("$powerUpsFolder/lives_up.png")

    override val duration: Int
        get() = 0

    override fun doApply(entity: BomberEntity) {
        DataInputOutput.getInstance().increaseLives()
    }

    override fun cancel(entity: BomberEntity) {
        // No need to implement anything here
    }

    override val isDisplayable: Boolean
        get() = false

    override fun canPickUp(entity: BomberEntity): Boolean = true

    override val type: EntityTypes
        get() = EntityTypes.LivesPowerUp
}
